### PR TITLE
Replace action to setup minikube with medyagh/setup-minikube

### PR DESCRIPTION
**Pull request by tenzen-y**
**What this PR does / why we need it**:
I replaced actions to set up minikube with [medyagh/setup-minikube](https://github.com/medyagh/setup-minikube) since using manusa/actions-setup-minikube seems to no longer be maintained, and the action doesn't support kubernetes v1.26.

related to: https://github.com/kubeflow/katib/pull/2177

**Which issue(s) this PR fixes** _(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)_:
Fixes #

**Checklist:**

- [ ] [Docs](https://www.kubeflow.org/docs/components/katib/) included if any changes are user facing

### CODE REVIEW

File: .github/workflows/e2e-test-darts-cifar10.yaml [hunk 1]
```diff
@@ -9,9 +9,6 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
 
-env:
-  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
```
Comment (andreyvelich) [inline .github/workflows/e2e-test-darts-cifar10.yaml:14]: Did we require GitHub Token before for our E2Es?

Comment (tenzen-y) [reply .github/workflows/e2e-test-darts-cifar10.yaml:14]: Yes, the token is required for the `manusa/actions-setup-minikube`.

**Comment by tenzen-y** [comment]
This PR is ready for review. Also, I confirmed this change is effective for k8s v1.26 at  #2177.
/assign @andreyvelich @johnugeorge

**Review by gaocegege — APPROVED** [review]
LGTM

**Review by andreyvelich — APPROVED** [review]
Thank you for this update @tenzen-y!
/lgtm
/approve

**Pull request merged**